homeassistant/components/webostv/config_flow.py

```diff
@@ -0,0 +1,203 @@
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class FlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """WebosTV configuration flow."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize workflow."""
+        self._host = None
+        self._name = None
+        self._uuid = None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, import_info):
+        """Set the config entry up from yaml."""
+        self._host = import_info[CONF_HOST]
+        self._name = import_info.get(CONF_NAME) or import_info[CONF_HOST]
+        return await self.async_step_pairing()
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            self._host = user_input[CONF_HOST]
+            self._name = user_input[CONF_NAME]
+            return await self.async_step_pairing()
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_check_configured_entry(self):
+        """Check if entry is configured, update unique_id if needed."""
+        for entry in self._async_current_entries(include_ignore=False):
+            if entry.data[CONF_HOST] != self._host:
+                continue
+
+            if self._uuid and not entry.unique_id:
+                _LOGGER.debug(
+                    "Updating unique_id for host %s, unique_id: %s",
+                    self._host,
+                    self._uuid,
+                )
+                self.hass.config_entries.async_update_entry(entry, unique_id=self._uuid)
+
+            return True
+
+        return False
+
+    async def async_step_pairing(self, user_input=None):
+        """Display pairing form."""
+        if await self.async_check_configured_entry():
+            return self.async_abort(reason="already_configured")
+
+        self.context[CONF_HOST] = self._host
+        self.context["title_placeholders"] = {"name": self._name}
+        errors = {}
+
+        if (
+            self.context["source"] == config_entries.SOURCE_IMPORT
+            or user_input is not None
+        ):
+            try:
+                client = await async_control_connect(self._host, None)
+            except PyLGTVPairException:
+                return self.async_abort(reason="error_pairing")
+            except WEBOSTV_EXCEPTIONS:
+                errors["base"] = "cannot_connect"
+            else:
+                data = {CONF_HOST: self._host, CONF_CLIENT_SECRET: client.client_key}
+                return self.async_create_entry(title=self._name, data=data)
```

Comment:
How will we handle the manually configured config entries that don't set a unique id? Won't they potentially be discovered again?

Reply:
When importing entries from yaml no unique id is set to the created entry, but host field is still used to prevent adding duplicates. Upon ssdp message the UDN is set as unique id and entries without unique id are updated with the id from the UDN if host match the entry host.

Reply:
I don't see where we're matching existing entry host data against new entry host data and aborting. We have this helper for that:
https://github.com/home-assistant/core/blob/0d38ee7378b6c8f3e3e34326e1eec3bbd8460db6/homeassistant/config_entries.py#L1141-L1145

Reply:
This checks if a new entry host match existing one:
https://github.com/home-assistant/core/blob/f9734616a23cf5c076b5fc68527a1431caf39c84/homeassistant/components/webostv/config_flow.py#L72-L88

It is used to update `unique_id` to existing entity and if it returns true pairing is aborted:
https://github.com/home-assistant/core/blob/f9734616a23cf5c076b5fc68527a1431caf39c84/homeassistant/components/webostv/config_flow.py#L92-L93